File: week1C/problemSet1/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

float get_valid_float(void);
int calculateCoins(int owedCashCoins);

int main(void)
{
  // ask user the change (float)
  float owedCash = get_valid_float();
  // convert dollars to cents float to int
  int owedCashCoins = round(owedCash * 100);
  // Do the math Loop reducing the coins from the owed value
  int coins = calculateCoins(owedCashCoins);
  // return the number of coins to return
  printf("%i\n", coins);
}

int calculateCoins(int owedCashCoins)
{
  // declare the coins
  int quarters = 25;
  int dimes = 10;
  int nickels = 5;
  int pennies = 1;
  int totalCoins = 0;
  do
  {
    if (owedCashCoins >= quarters)
    {
      owedCashCoins -= quarters;
      totalCoins++;
    }
    else if (owedCashCoins >= dimes)
    {
      owedCashCoins -= dimes;
      totalCoins++;
    }
    else if (owedCashCoins >= nickels)
    {
      owedCashCoins -= nickels;
      totalCoins++;
    }
    else if (owedCashCoins >= pennies)
    {
      owedCashCoins -= pennies;
      totalCoins++;
    }
  } while (owedCashCoins > 0);
  return totalCoins;
}

float get_valid_float(void)
{
  float n;
  do
  {
    n = get_float("Change owed: ");
  } while (n < 0);
  return n;
}

File: week6Python/pset6/cash/cash.py
from cs50 import get_float

while True:
    owedCashF = get_float("Change owed: ")
    if owedCashF > 0:
        break

owedCash = owedCashF * 100
quarters = 25;
dimes = 10;
nickels = 5;
pennies = 1;
totalCoins = 0;

while True:
    if owedCash <= 0:
        break
    if owedCash >= quarters:
        owedCash -= quarters
        totalCoins += 1
    elif owedCash >= dimes:
        owedCash -= dimes
        totalCoins += 1
    elif owedCash >= nickels:
        owedCash -= nickels
        totalCoins += 1
    else:
        owedCash -= pennies
        totalCoins += 1

print(f"{totalCoins}")